pkg/controller/nexus/resource/deployment.go

```diff
@@ -117,12 +120,38 @@ func addProbes(deployment *appsv1.Deployment) {
 				Scheme: v1.URISchemeHTTP,
 			},
 		},
-		InitialDelaySeconds: 60,
-		TimeoutSeconds:      10,
+		InitialDelaySeconds: probeInitialDelaySeconds,
+		TimeoutSeconds:      probeTimeoutSeconds,
 	}
 
-	deployment.Spec.Template.Spec.Containers[0].ReadinessProbe = defaultProbe
-	deployment.Spec.Template.Spec.Containers[0].LivenessProbe = defaultProbe
+	deployment.Spec.Template.Spec.Containers[0].LivenessProbe = defaultProbe.DeepCopy()
+	deployment.Spec.Template.Spec.Containers[0].ReadinessProbe = defaultProbe.DeepCopy()
+
+	if nexus.Spec.LivenessProbe != nil {
+		deployment.Spec.Template.Spec.Containers[0].LivenessProbe.FailureThreshold =
+			util.EnsureMinimum(nexus.Spec.LivenessProbe.FailureThreshold, 1)
+		deployment.Spec.Template.Spec.Containers[0].LivenessProbe.InitialDelaySeconds =
+			util.EnsureMinimum(nexus.Spec.LivenessProbe.InitialDelaySeconds, 0)
+		deployment.Spec.Template.Spec.Containers[0].LivenessProbe.PeriodSeconds =
+			util.EnsureMinimum(nexus.Spec.LivenessProbe.PeriodSeconds, 1)
+		deployment.Spec.Template.Spec.Containers[0].LivenessProbe.SuccessThreshold =
+			util.EnsureMinimum(nexus.Spec.LivenessProbe.SuccessThreshold, 1)
+		deployment.Spec.Template.Spec.Containers[0].LivenessProbe.TimeoutSeconds =
+			util.EnsureMinimum(nexus.Spec.LivenessProbe.TimeoutSeconds, 1)
+	}
+
+	if nexus.Spec.ReadinessProbe != nil {
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.FailureThreshold =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.FailureThreshold, 1)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.InitialDelaySeconds =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.InitialDelaySeconds, 0)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.PeriodSeconds =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.PeriodSeconds, 1)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.SuccessThreshold =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.SuccessThreshold, 1)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.TimeoutSeconds =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.TimeoutSeconds, 1)
+	}
```

Comment:
We don't need to validate in the Reconcile Loop since we're enforcing this check in the OpenAPI:  https://github.com/m88i/nexus-operator/pull/68/files#diff-115e08ad98fca2d2d354114263813118R139

(this is where we will have calls in the loop)

By adding this check on deployment creation, I can enforce the check at the API level in case we have an external call from other functions/internal packages. This is not an interface validation, but a sanity check for the k8s API integration.

Having a TLS check in the reconcile loop makes sense because we can't enforce that in the OpenAPI (external interface) level. But I think we could add this check in the resource creation as well (or better, move it to there).

To sum up, interface checks we do in the reconcile, internal calls validations we do in the package/internal level.

Reply:
I see, that makes sense! I hadn't taken into account API usage, I was only considering usage from the reconcile loop point of view.